firebase-common/src/main/java/com/google/firebase/emulators/FirebaseEmulators.java

```diff
@@ -0,0 +1,34 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.firebase.emulators;
+
+import com.google.android.gms.common.annotation.KeepForSdk;
+
+/**
+ * Enum for Firebase services that can be emulated using the Firebase Emulator Suite.
+ *
+ * <p>TODO(samstern): Un-hide this once Firestore, Database, and Functions are implemented
+ *
+ * @see com.google.firebase.FirebaseApp#enableEmulators(EmulatorSettings)
+ * @see EmulatorSettings
+ * @see EmulatedServiceSettings
+ * @hide
+ */
+@KeepForSdk
+public enum FirebaseEmulators {
```

Comment:
While making this an enum makes the API simple and type safe, it couples firebase-common to all baas sdks.

To avoid the coupling I suggest to either resort to using `String`s with each string defined as a constant on the target SDK, i.e. `FirebaseFirestore.EMULATOR = "firestore"` or something.
Or wrapping that string in a class here in common like `data class Emulator(serviceName: String)` and initializing the SDK constant with it instead of the string.

A potential additional advantage of this is that one can only configure an emulator for say `firestore` if it's actually present in their app.